design/20210209.certificates.k8s.io-adoption.md

```diff
@@ -0,0 +1,157 @@
+resource graduated to `certificates.k8s.io/v1`. This makes the concept of
+requesting, signing, and consuming certificates in Kubernetes first class
+objects. cert-manager is well placed to serve this resource type as it becomes
+more popular in the wider community, whilst preserving all the features and
+extensions that cert-manager has to offer.
+
+
+## Motivation
+
+With the extensiveness of cert-manager’s ecosystem, it is well placed to manage
+the core Kubernetes resource type. This gives cert-manager the ability to
+integrate with components that it otherwise wouldn't have, without changes to the
+third-party project.
+
+
+### Goals
+
+- Add `CertificateSigningRequest` signer controllers to cert-manager for all
+    `Issuer` types
+
+### Non-Goals
+
+- Remove the `CertificateRequest` resource in favour of
+    `CertificateSigningRequest`
+- Change the `Certificate` controllers behaviour
+- Support `CertificateSigningRequest` pre v1
+- Make changes to upstream Kubernetes to implement controllers in cert-manager
+
+
+## Reasons for Approach
+
+Below are a summary of properties of the `CertificateSigningRequest`, and
+explanations as to why other approaches to integrate the resource with
+cert-manager aren't possible (i.e. completely switching the `CertificateRequest`
+resource to `CertificateSigningRequest`:
+
+1. `CertificateSigningRequests` are cluster scoped resources. This means that
+   the concept of namespaced `Issuers` doesn't fit with this resource, and any
+   attempt to integrate it would open up the ability of privilege escalation
+   (i.e. referencing Secrets in another tenants Namespace). Switching to
+   `CertificateSigningRequests` would mean the removal of the `Issuer` resource.
+
+1. As `CertificateSigningRequests` are cluster scoped, they do not provide the
+   same kind of isolation that `CertificateRequests` provide- preventing tenants
+   reading others requests, namespace specific issuer authentication (e.g. Vault
+   Kubernetes auth), namespacing cert-manager.
+
+1. `CertificateSigningRequests` do not include a duration field. This means that
+   any attempt to "proxy", or copy `CertificateRequests`, would result in a
+   regression of this feature. A feature which many users of cert-manager rely
+   on.
+
+1. cert-manager has a large number of [external
+   issuers](https://cert-manager.io/docs/configuration/external/) created by
+   the community that operate over the `CertificateRequest` resource. No longer
+   supporting these resources would require these authors to completely re-write
+   these issuers to continue to be supported by cert-manager.
+
+Supporting both resource types, but making `CertificateRequests` the "priority"
+resource (i.e. the resource created by the `Certificates` controller), means
+that we don't have any regression in the features cert-manager offers, whilst
+maintaining the same security properties.
```

Comment:
I'm not too sure if this is an approach we want to go - we are effectively forking the community of API consumers 🤔 how do we expect this to work long term?